client.go

```diff
@@ -0,0 +1,727 @@
+func (c *Client) Rpop(key string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		value, err := c.redisClient.RPop(c.vu.Context(), key).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(value)
+	}()
+
+	return promise
+}
+
+// Lrange returns the specified elements of the list stored at `key`. The
+// offsets start and stop are zero-based indexes. These offsets can be
+// negative numbers, where they indicate offsets starting at the end of
+// the list.
+func (c *Client) Lrange(key string, start, stop int64) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		values, err := c.redisClient.LRange(c.vu.Context(), key, start, stop).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(values)
+	}()
+
+	return promise
+}
+
+// Lindex returns the specified element of the list stored at `key`.
+// The index is zero-based. Negative indices can be used to designate
+// elements starting at the tail of the list.
+func (c *Client) Lindex(key string, index int64) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		value, err := c.redisClient.LIndex(c.vu.Context(), key, index).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(value)
+	}()
+
+	return promise
+}
+
+// Lset sets the list element at `index` to `element`.
+func (c *Client) Lset(key string, index int64, element string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		value, err := c.redisClient.LSet(c.vu.Context(), key, index, element).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(value)
+	}()
+
+	return promise
+}
+
+// Lrem removes the first `count` occurrences of `value` from the list stored
+// at `key`. If `count` is positive, elements are removed from the beginning of the list.
+// If `count` is negative, elements are removed from the end of the list.
+// If `count` is zero, all elements matching `value` are removed.
+func (c *Client) Lrem(key string, count int64, value string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		n, err := c.redisClient.LRem(c.vu.Context(), key, count, value).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(n)
+	}()
+
+	return promise
+}
+
+// Llen returns the length of the list stored at `key`. If `key`
+// does not exist, it is interpreted as an empty list and 0 is returned.
+func (c *Client) Llen(key string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		length, err := c.redisClient.LLen(c.vu.Context(), key).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(length)
+	}()
+
+	return promise
+}
+
+// Hset sets the specified field in the hash stored at `key` to `value`.
+// If the `key` does not exist, a new key holding a hash is created.
+// If `field` already exists in the hash, it is overwritten.
+func (c *Client) Hset(key, field, value string) *goja.Promise {
```

Comment:
shouldn't `value` here be an `interface{}`

Reply:
indeed 😓